tiledb/sm/cpp_api/deleter.h

```diff
@@ -134,6 +134,10 @@ class Deleter {
     tiledb_consolidation_plan_free(&p);
   }
 
+  void operator()(tiledb_string_t* p) const {
+    tiledb_string_free(&p);
```

Comment:
This function has a return value that should not be ignored. It's a low-likelihood failure, but at least it deserves a log message if it fails.
Other handles in the C API are going to have `_free` functions that will return (with an appropriate change cycle), so it would be appropriate to add a generic log message.

Reply:
Does the C++ API have such facility to log messages?

Reply:
(reverted it, it was causing errors, let's decide what to do first)

Reply:
Apparently the C++ API does not have a logging facility. That's a problem.

Reply:
I added a stub logging class, it likely needs improvements.